vuework04/src/firebase/firebase.config.js

```diff
@@ -1,18 +1,36 @@
 import firebase from 'firebase/app';
 import 'firebase/auth';
-import store from '../store';
+import 'firebase/firestore';
+import store from '../store/index';
 const firebaseConfig = {
   apiKey: 'AIzaSyC7shKU7YhltXKuKZgFZmANxxKJ7bS8wM0',
-  authDomain: 'PROJECT_ID.firebaseapp.com',
-  databaseURL: 'https://PROJECT_ID.firebaseio.com',
-  projectId: 'PROJECT_ID',
-  storageBucket: 'PROJECT_ID.appspot.com',
-  messagingSenderId: 'SENDER_ID',
+  authDomain: 'vuework04.firebaseapp.com',
+  projectId: 'vuework04',
+  storageBucket: 'vuework04.appspot.com',
+  messagingSenderId: '316792669101',
   appId: '1:316792669101:web:b2395e000f3910b8331724',
-  measurementId: 'G-MEASUREMENT_ID',
+  databaseURL: 'https://vuework04.firebaseio.com',
 };
+
 firebase.initializeApp (firebaseConfig);
+firebase.firestore ();
 
+//ログイン状態の確認
 firebase.auth ().onAuthStateChanged (user => {
-  store.commit ('addUser', user);
+  if (user) {
+    console.log (user);
+    user.providerData.forEach (function (profile) {
+      // ログインしているユーザ情報をストアにコミット
+      store.dispatch ('setLoginUser', user);
+
+      //現在ログインしているユーザの確認のため、コンソールに出力
+      console.log ('Sign-in provider: ' + profile.providerId);
+      console.log ('  Provider-specific UID: ' + profile.uid);
+      console.log ('  Name: ' + profile.displayName);
+      console.log ('  Email: ' + profile.email);
+      console.log ('  wallet: ' + profile.wallet);
+    });
+  } else {
+    console.log ('ログインしてください');
+  }
```

Comment:
①ログイン判定をしてユーザ情報をstoreに格納する役割ですが、configに記述するのは違和感があります。ダッシュボードに記述するのが自然ですね。

②`providerData`というのはプロバイダー（SNSなど）を使ってログインした際に、そのプロバイダに登録のあるデータを引っ張ってくる時に使います。ですので今回の場合は下記のようになりますね。
```suggestion
  if (user) {
    console.log (user);
      // ログインしているユーザ情報をストアにコミット
      store.dispatch ('setLoginUser', user);

      //現在ログインしているユーザの確認のため、コンソールに出力
      console.log ('  Name: ' + user.displayName);
      console.log ('  Email: ' + user.email);
      console.log ('  wallet: ' + user.wallet);
  } else {
    console.log ('ログインしてください');
  }
```
今回は課題ですのでOKですが、現場では挙動確認用のコードなど要件に関係のないコードはpushするのはNGになっていることを覚えておきましょう。